service-test/test.ml

```diff
@@ -0,0 +1,129 @@
+open Letters
+
+let ( let* ) = Lwt.bind
+
+let get_ethereal_account_details () =
+  let open Yojson.Basic.Util in
+  (* see the README.md how to generate the account file and the path
+   * below is relative to the location of the executable under _build
+   *)
+  let json = Yojson.Basic.from_file "../../../ethereal_account.json" in
+  let username = json |> member "user" |> to_string in
+  let password = json |> member "pass" |> to_string in
+  let smtp_node = json |> member "smtp" in
+  let hostname = smtp_node |> member "host" |> to_string in
+  let port = smtp_node |> member "port" |> to_int in
+  let with_starttls = smtp_node |> member "secure" |> to_bool |> not in
+  Lwt.return
+    {
+      sender = username;
+      username;
+      password;
+      hostname;
+      port = Some port;
+      with_starttls;
+      ca_dir = "/etc/ssl/certs";
+    }
+
+let test_send_plain_text_email config _ () =
+  let recipients =
+    [
+      To "harry@example.com";
+      To "larry@example.com";
+      Cc "bill@example.com";
+      Bcc "dave@example.com";
+    ]
+  in
+  let subject = "Plain text test email" in
+  let body =
+    Plain {|
+Hi there,
+
+have you already seen the very cool new web framework written in ocaml:
+https://github.com/oxidizing/sihl
+
+Regards,
+The team
+|}
+  in
+  let message = build_email ~from:config.sender ~recipients ~subject ~body in
+  let* res = send ~config ~recipients ~message in
+  match res with Ok () -> Lwt.return () | Error msg -> failwith msg
+
+let test_send_html_email config _ () =
+  let recipients =
+    [
+      To "harry@example.com";
+      To "larry@example.com";
+      Cc "bill@example.com";
+      Bcc "dave@example.com";
+    ]
+  in
+  let subject = "HTML only test email" in
+  let body =
+    Html {|
+<p>Hi there,</p>
+<p>
+  have you already seen the very cool new web framework written in ocaml:
+  <a href="https://github.com/oxidizing/sihl">Sihl</a>
+<p>
+Regards,<br>
+The team
+</p>
+|}
+  in
+  let message = build_email ~from:config.sender ~recipients ~subject ~body in
+  let* res = send ~config ~recipients ~message in
+  match res with Ok () -> Lwt.return () | Error msg -> failwith msg
+
+let test_send_mixed_body_email config _ () =
+  let recipients =
+    [
+      To "harry@example.com";
+      To "larry@example.com";
+      Cc "bill@example.com";
+      Bcc "dave@example.com";
+    ]
+  in
+  let subject = "Mixed body email with plain text and HTML" in
+  let text = {|
+Hi there,
+
+have you already seen the very cool new web framework written in ocaml:
+https://github.com/oxidizing/sihl
+
+Regards,
+The team
+|}
+  in
+  let html = {|
+<p>Hi there,</p>
+<p>
+  have you already seen the very cool new web framework written in ocaml:
+  <a href="https://github.com/oxidizing/sihl">Sihl</a>
+<p>
+Regards,<br>
+The team
+</p>
+|}
+  in
+  let message = build_email ~from:config.sender ~recipients ~subject ~body:(Mixed (text, html, None)) in
```

Comment:
Why do we expose the boundary through the API?

Reply:
It's optional and needed so that we can actually test the feature better.

Reply:
Imo this is too low level to expose it through the client API.